app/models/zone.rb

```diff
@@ -261,4 +262,8 @@ def check_zone_in_use_on_destroy
     msg = message_for_invalid_delete
     raise msg if msg
   end
+
+  def clear_my_region_zone_cache
+    MiqRegion.my_region.clear_my_relation_cache
```

Comment:
It might be cleaner / simpler to just bust the region itself...then you also don't need to expose the clear_my_relation_cache